nifi-extension-bundles/nifi-websocket-bundle/nifi-websocket-services-jetty/src/main/java/org/apache/nifi/websocket/jetty/JettyWebSocketServer.java

```diff
@@ -254,6 +261,21 @@ public Object createWebSocket(JettyServerUpgradeRequest servletUpgradeRequest, J
         }
     }
 
+    private static int getPort(JettyServerUpgradeRequest servletUpgradeRequest) {
+        final Object localSocketAddress = servletUpgradeRequest.getLocalSocketAddress();
+        if (localSocketAddress == null) {
+            throw new RuntimeException("Local socket address is null");
+        }
+
+        // Check if the local socket address is an instance of InetSocketAddress
+        if (!(localSocketAddress instanceof InetSocketAddress)) {
+            throw new RuntimeException("Local socket address is not an instance of InetSocketAddress: " + localSocketAddress.getClass().getName());
+        }
+
+        final int port = ((InetSocketAddress) localSocketAddress).getPort();
+        return port;
```

Comment:
This could be consolidated with the above check using `instance of InetSocketAddres socketAddress` and then using the named `socketAddress` variable.